oak-run-commons/src/main/java/org/apache/jackrabbit/oak/index/IndexerSupport.java

```diff
@@ -121,7 +121,7 @@ public void updateIndexDefinitions(NodeBuilder rootBuilder) throws IOException,
 
     protected void dumpIndexDefinitions(NodeStore nodeStore) throws IOException {
         IndexDefinitionPrinter printer = new IndexDefinitionPrinter(nodeStore, indexHelper.getIndexPathService());
-        printer.setFilter("{\"properties\":[\"*\", \"-:childOrder\"],\"nodes\":[\"*\", \"-:index-definition\"]}");
+        printer.setFilter("{\"properties\":[\"*\", \"-:childOrder\"],\"nodes\":[\"*\", \"-:index-definition\", \"-:data\", \"-:suggest-data\"]}");
```

Comment:
I'm not sure if this is actually correct.

Isn't the dump needed for import? If yes, then that would mean if we don't export the ":data" node, import of an index would have to do a full reindex as the index data is empty...

So, first we need to check where this dumpIndexDefinitions method is actually called from... unluckily, there is no documentation it seems.

Reply:
I tested it locally and found out that index definitions don't have :data if --index-paths are provided in command line. But it seems to have :data if --index-paths are not provided.
For :suggest-data also, found out that :suggest-data node only contained   "lastUpdated" property in case --index-paths are provided. So looks like reindexing works without this info.

As :status-data is something which was part of index-definitions, I added a test case just to check if suggest data is preset with this new filter. Let me know if you think we should add few more test cases/ any specific case you think we should incorporate in current test.

Reply:
Hi @thomasmueller , 
Looks like we don't need hidden nodes at all in index-definitions file. While importing index we do cloneVisibleState from index definition.